src/encoder/mod.rs

```diff
@@ -0,0 +1,494 @@
+use std::io::{Write, Seek};
+use std::collections::BTreeMap;
+use byteorder::{NativeEndian};
+
+use crate::decoder::ifd::{self, Tag};
+use crate::error::{TiffResult, TiffFormatError, TiffError};
+
+mod writer;
+pub mod colortype;
+
+use self::writer::*;
+use self::colortype::*;
+
+/// Type to represent tiff values of type `RATIONAL`
+#[derive(Clone)]
+pub struct Rational {
+    pub n: u32,
+    pub d: u32,
+}
+
+/// Type to represent resolution units
+pub enum ResolutionUnit {
+    None = 1,
+    Inch = 2,
+    Centimeter = 3,
+}
+
+/// Trait for types that can be encoded in a tiff file
+pub trait TiffValue {
+    const BYTE_LEN: u32;
+    const FIELD_TYPE: ifd::Type;
+    fn count(&self) -> u32;
+    fn bytes(&self) -> u32 {
+        self.count() * Self::BYTE_LEN
+    }
+    fn write<W: Write>(&self, writer: &mut TiffWriter<W>) -> TiffResult<()>;
+}
+
+impl TiffValue for &[u8] {
```

Comment:
Is there a reason why this is implemented for `&'_ [u8]` instead of `[u8]`? All required methods take `&self` and thus `Sized` is not required. The `impl` for `str` is as expected.